### PR TITLE
Bugfix: Accessibility.open init from token

### PR DESCRIPTION
**Pull request by aim2120**
## Description

The `Accessibility` initializer is not properly initializing the `open` case when for the version of swift-syntax used by the project. This was resulting in open class mocks being generated with the default (internal) access level.

### Bug

With the library's current behavior, this code:
```swift
open class MyOpenClass {
  open func myFunc() { }
}
```
would generate this mock (simplified for readability):
```swift
class MockMyOpenClass {
  override func myFunc() { ... }
}
```

### Fix

Match `.keyword(.open)` instead of `.identifier("open")` for the `open` access level.

With this change, the hypothetical class would be generated as:
```swift
public class MockMyOpenClass {
  public override func myFunc() { ... }
}
```

### Tests

To create access level separation between mocks and test code, I've created `CuckooMocks` frameworks for each platform, each depended upon by each corresponding test target.  The only change to existing tests is that this framework must be imported. More importantly, this separation allows for tests to prove that access levels of mocks are properly generated.

I'm open to feedback from the maintainer if they have differences in preference in this project structuring. If there were a way to determine access level from another method (e.g. reflection), I would opt for that, but it seems that this is really the only way with Swift to prove `internal` vs. `public` access levels for generated mocks.

### CODE REVIEW

**Review by MatyasKriz — APPROVED** [review]
Great work with this addition! There were ambitions of creating a separate target with mocks to mimic the same access users' unit tests have, but I'd never expect it to be so straight-forward with the help of Tuist. Thank you. 🙂

I'll take a look at this on my machine and merge it as soon as I'm done.

File: Project.swift [hunk 1]
```diff
@@ -32,16 +37,16 @@ func platformSet(platform: PlatformType) -> (targets: [Target], schemes: [Scheme
     )
     targets.append(defaultTarget)
 
-    let defaultTestTarget = Target(
-        name: "Cuckoo-\(platform)Tests",
+    let mocksTarget = Target(
+        name: "CuckooMocks-\(platform)",
```
[the remark this answers is not in the document]
Comment (aim2120) [reply Project.swift:41]: Sure!

[the remark this answers is not in the document]
Comment (aim2120) [reply Project.swift:41]: Do you still like `CuckooMocks` as the product name for importing? See Project.swift L9. Or would you prefer something like `CuckooTestMocks`?

File: Project.swift [hunk 2]
```diff
@@ -5,11 +5,16 @@ let commonBuildSettingsBase: [String: SettingValue] = [
...
+    "PRODUCT_NAME": .string("CuckooMocks"),
+    "SYSTEM_FRAMEWORK_SEARCH_PATHS": .string("$(PLATFORM_DIR)/Developer/Library/Frameworks"),
+]
 let objCBuildSettingsBase: [String: SettingValue] = [
     "SWIFT_OBJC_BRIDGING_HEADER": .string("$(PROJECT_DIR)/OCMock/Cuckoo-BridgingHeader.h"),
 ]
 
 let defaultBuildSettings = Settings.settings(base: commonBuildSettingsBase)
+let mocksBuildSettings = Settings.settings(base: mocksBuildSettingsBase)
 let objCBuildSettings = Settings.settings(base: commonBuildSettingsBase.merging(objCBuildSettingsBase, uniquingKeysWith: { $1 }))
```
[the remark this answers is not in the document]
Comment (aim2120) [reply Project.swift:18]: Done!

File: Project.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatyasKriz) [reply Project.swift]: The current name is good. 🙂

**Pull request closed without merging**